### PR TITLE
Feature/aa 264 deactivate deleted accommodations

### PR DESCRIPTION
**Pull request by hasmikharutyunyan**

### CODE REVIEW

**Review by oleg1std — APPROVED** [review]

File: HappyTravel.Nakijin.Api/Services/Workers/AccommodationMapping/AccommodationMapper.cs [hunk 1]
```diff
@@ -250,11 +259,16 @@ public class AccommodationMapper : IAccommodationMapper
 
             _context.AddRange(accommodationsToAdd);
             _context.AddRange(uncertainAccommodationsToAdd);
+            _context.AddRange(htAccommodationMappingsToAdd);
             await _context.SaveChangesAsync(cancellationToken);
 
             var accommodationsToPublish = accommodationsToAdd
                 .Where(a => a.IsActive)
-                .Union(accommodationsFromUncertainToPublish);
+                .Union(accommodationsFromUncertainToPublish).ToList();
```
Comment (kirillta) [inline HappyTravel.Nakijin.Api/Services/Workers/AccommodationMapping/AccommodationMapper.cs:267]: `ToList()` expects on aa new line

File: HappyTravel.Nakijin.Api/Services/Workers/AccommodationMapping/AccommodationMapper.cs [hunk 2]
```diff
@@ -313,21 +327,39 @@ void AddUncertain(Contracts.MultilingualAccommodation accommodation, int existin
             }
 
 
-            void AddOrChangeActivity(Contracts.MultilingualAccommodation accommodation, bool isActive, 
-                DeactivationReasons deactivationReason = DeactivationReasons.None)
+            void ProcessDeactivatedAccommodation(Contracts.MultilingualAccommodation accommodation)
+            {
+                if (activeCountryAccommodationsOfSupplier.TryGetValue(accommodation.SupplierCode, out var existingAccommodation))
```
Comment (kirillta) [inline HappyTravel.Nakijin.Api/Services/Workers/AccommodationMapping/AccommodationMapper.cs:332]: This statement could be reversed

File: HappyTravel.Nakijin.Api/Services/Workers/AccommodationMapping/AccommodationMapper.cs [hunk 3]
```diff
@@ -313,21 +327,39 @@ void AddUncertain(Contracts.MultilingualAccommodation accommodation, int existin
...
+                {
+                    if (existingAccommodation.SupplierAccommodationCodes.Count > 1)
+                    {
+                        var dbAccommodation = GetDbAccommodation(accommodation, true);
```
Comment (kirillta) [inline HappyTravel.Nakijin.Api/Services/Workers/AccommodationMapping/AccommodationMapper.cs:336]: Please, specify the second parameter

File: HappyTravel.Nakijin.Api/Services/Workers/AccommodationMapping/AccommodationMapper.cs [hunk 4]
```diff
@@ -313,21 +327,39 @@ void AddUncertain(Contracts.MultilingualAccommodation accommodation, int existin
...
+                        foreach (var supplierCode in existingAccommodation.SupplierAccommodationCodes)
+                            dbAccommodation.SupplierAccommodationCodes.TryAdd(supplierCode.Key, supplierCode.Value);
+
+                        dbAccommodation.SupplierAccommodationCodes.Remove(supplier);
+
+                        AddOrUpdateHtAccommodationMappings(existingAccommodation.HtId, 0, dbAccommodation);
```
Comment (kirillta) [inline HappyTravel.Nakijin.Api/Services/Workers/AccommodationMapping/AccommodationMapper.cs:342]: What 0 means here?

Comment (hasmikharutyunyan) [reply HappyTravel.Nakijin.Api/Services/Workers/AccommodationMapping/AccommodationMapper.cs]: I have changed to named parameter. 0 means that needed create new accommodation. I have done this logic to not duplicate code.

**Review by kirillta — APPROVED** [review]

**Pull request merged**